File: ProyectoBaseWebApp/app/OrigenMadera.php
<?php

namespace App;

use Illuminate\Database\Eloquent\Model;

class OrigenMadera extends Model
{
    protected $table = 'origenes_madera';
    protected $fillable = [
        'empresa_id', 'descripcion', 'anio_cultivo', 'volumen_inventario', 'hectareas', 'estado', 'creador_id', 'modificador_id'
    ];
    public function scopeActive($query)
    {
        return $query->where('estado', 'A');
    }
    public function creador()
    {
        return $this->belongsTo('App\User');
    }
    public function modificador()
    {
        return $this->belongsTo('App\User');
    }
    public function scopeCurrent($query, $empresaId)
    {
        return $query->where('empresa_id', $empresaId);
    }
}

File: CosechasApp/app/src/main/java/com/techtraining/cosechasapp/db/OrigenMadera.java
package com.techtraining.cosechasapp.db;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity
public class OrigenMadera {
    @PrimaryKey
    public int id;

    @ColumnInfo(name = "descripcion")
    public String descripcion;

    @ColumnInfo(name = "anio_cultivo")
    public int anioCultivo;

    @ColumnInfo(name = "volumen_inventario")
    public double volumenInventario;

    @ColumnInfo(name = "hectareas")
    public double hectareas;

    @ColumnInfo(name = "empresa_id")
    public int empresaId;

    @ColumnInfo(name = "estado")
    public String estado;

    @Override
    public String toString() {
        return descripcion + " (" + this.anioCultivo + ")";
    }
}
